src/providers/ruby.rs

```diff
@@ -0,0 +1,95 @@
+use super::{node::NodeProvider, Provider};
+use crate::nixpacks::{
+    app::App,
+    environment::Environment,
+    nix::pkg::Pkg,
+    phase::{InstallPhase, SetupPhase, StartPhase},
+};
+use anyhow::{Ok, Result};
+pub struct RubyProvider {}
+
+impl Provider for RubyProvider {
+    fn name(&self) -> &str {
+        "Ruby"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("Gemfile"))
+    }
+
+    fn setup(&self, app: &App, env: &Environment) -> Result<Option<SetupPhase>> {
+        let mut pkgs = vec![Pkg::new("gcc"), self.get_nix_ruby_package(app)];
+        if app.includes_file("package.json") {
+            pkgs.push(NodeProvider::get_nix_node_pkg(
+                &app.read_json("package.json")?,
+                env,
+            )?);
+        }
+        let setup_phase = SetupPhase::new(pkgs);
+        Ok(Some(setup_phase))
+    }
+
+    fn install(&self, app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
+        let mut install_phase = InstallPhase::new(format!(
+            "gem install {} && bundle install",
+            self.get_bundler_version(app)
+        ));
+        install_phase.add_file_dependency("Gemfile".to_string());
+        if app.includes_file("Gemfile.lock") {
+            install_phase.add_file_dependency("Gemfile.lock".to_string());
+        }
+        if app.includes_file("package.json") {
+            install_phase.add_file_dependency("package.json".to_string());
+            install_phase.cmd = Some(format!("npm install && {}", install_phase.cmd.unwrap()));
+        }
+        Ok(Some(install_phase))
+    }
+
+    fn start(&self, _app: &App, _env: &Environment) -> Result<Option<StartPhase>> {
+        Ok(None)
+    }
+}
+
+impl RubyProvider {
+    // fn detect_framework(&self, app: &App) -> String {
```

Comment:
Just quick FYI, this should work by detecting gems in the gemfile.

Reply:
Not the best way to detect gems. Current logic is based on how buildpacks work. So, it should be fine.

Reply:
What is a better way? That is what we decided on.

Reply:
Current logic is better because buildpacks have used that for years now.